File: apps/ios/Packages/Model/Sources/Model/status/MediaServerType.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
// Seerr server/constants/server.ts at 69f73a6f1486fdb51b8ddae9a94a8dfb629f461c.
public enum MediaServerType: Sendable, Equatable, Codable {
    case plex
    case jellyfin
    case emby
    case notConfigured
    case unrecognized(Int?)

    public init(rawValue: Int?) {
        switch rawValue {
        case 1: self = .plex
        case 2: self = .jellyfin
        case 3: self = .emby
        case 4: self = .notConfigured
        default: self = .unrecognized(rawValue)
        }
    }

    public var rawValue: Int? {
        switch self {
        case .plex: 1
        case .jellyfin: 2
        case .emby: 3
        case .notConfigured: 4
        case .unrecognized(let raw): raw
        }
    }
}

File: apps/android/core/model/src/main/kotlin/app/gauja/core/model/status/MediaServerType.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.model.status

// Seerr server/constants/server.ts at 69f73a6f1486fdb51b8ddae9a94a8dfb629f461c.
sealed class MediaServerType(val rawValue: Int?) {
    data object PLEX : MediaServerType(1)

    data object JELLYFIN : MediaServerType(2)

    data object EMBY : MediaServerType(3)

    data object NOT_CONFIGURED : MediaServerType(4)

    data class Unrecognized(val value: Int?) : MediaServerType(value)

    companion object {
        fun fromWire(raw: Int?): MediaServerType =
            when (raw) {
                1 -> PLEX
                2 -> JELLYFIN
                3 -> EMBY
                4 -> NOT_CONFIGURED
                else -> Unrecognized(raw)
            }
    }
}
